saleor/graphql/core/connection.py

```diff
@@ -39,6 +42,25 @@ def get_field_value(instance: DjangoModel, field_name: str):
     return attr
 
 
+def _prepare_filter_by_rank_expression(
+    cursor: List[str],
+    sorting_direction: str,
+) -> Q:
+    try:
+        rank = Decimal(cursor[0])
+        int(cursor[1])
+    except (InvalidOperation, ValueError, TypeError, KeyError):
+        raise ValueError("Invalid cursor for sorting by rank.")
+
+    if sorting_direction == "gt":
+        return Q(rank__range=(rank - EPSILON, rank + EPSILON), id__lt=cursor[1]) | Q(
```

Comment:
Maybe we could add some line comments in this function to describe the logic here, what is EPSILON etc?